### ISSUE TITLE
panic

### ISSUE THREAD

**Issue by skoky** [DominoTree/rs-ipfix#2]
I see this in log when using the lib

```thread 'ipfix' panicked at 'range end index 297 out of range for slice of length 88', /usr/local/cargo/git/checkouts/rs-ipfix-42c2f7fc459fd94f/0714f22/src/lib.rs:130:38```

Is it anything obvious?

**Comment by DominoTree** [comment]
My first guess would be that the data it's being sent is somehow corrupted or doesn't match the IPFIX format. Do you happen to have a pcap of traffic that triggers this?

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by skoky** [comment]
yes, it was. I was passing in Netflow v9 starting with "0009" or even some other data. It would be wise to check netflow/ipfix version in header and also not to panic if length is invalid in "lib.rs:130"
Thanks for the lib anyway

**Comment by DominoTree** [comment]
I'll go ahead and make that change, it really shouldn't panic on any malformed or unexpected traffic
